datasets/flwr_datasets/partitioner/vertical_size_partitioner_test.py

```diff
@@ -0,0 +1,186 @@
+            VerticalSizePartitioner(partition_sizes=[5, -1])
+
+    def test_init_invalid_mode(self) -> None:
+        """Check ValueError if active_party_columns_mode is invalid."""
+        with self.assertRaises(ValueError):
+            VerticalSizePartitioner(
+                partition_sizes=[2, 2], active_party_columns_mode="invalid"
+            )
+
+    def test_init_active_party_column_invalid_type(self) -> None:
+        """Check ValueError if active_party_column is not str/list."""
+        with self.assertRaises(ValueError):
+            VerticalSizePartitioner(partition_sizes=[2, 2], active_party_column=123)
+
+    def test_partitioning_with_int_sizes(self) -> None:
+        """Check correct partitioning with integer sizes."""
+        columns = ["f1", "f2", "f3", "f4", "f5"]
+        dataset = _create_dummy_dataset(columns)
+        partitioner = VerticalSizePartitioner(partition_sizes=[2, 3], shuffle=False)
+        partitioner.dataset = dataset
+        p0 = partitioner.load_partition(0)
+        p1 = partitioner.load_partition(1)
+        self.assertEqual(len(p0.column_names), 2)
+        self.assertEqual(len(p1.column_names), 3)
+
+    def test_partitioning_with_fraction_sizes(self) -> None:
+        """Check correct partitioning with fraction sizes."""
+        columns = ["f1", "f2", "f3", "f4"]
+        dataset = _create_dummy_dataset(columns)
+        partitioner = VerticalSizePartitioner(partition_sizes=[0.5, 0.5], shuffle=False)
+        partitioner.dataset = dataset
+        p0 = partitioner.load_partition(0)
+        p1 = partitioner.load_partition(1)
+        self.assertEqual(len(p0.column_names), 2)
+        self.assertEqual(len(p1.column_names), 2)
+
+    def test_partitioning_with_drop_columns(self) -> None:
+        """Check dropping specified columns before partitioning."""
+        columns = ["f1", "drop_me", "f2", "f3"]
+        dataset = _create_dummy_dataset(columns)
+        partitioner = VerticalSizePartitioner(
+            partition_sizes=[2, 1], drop_columns=["drop_me"], shuffle=False
+        )
+        partitioner.dataset = dataset
+        p0 = partitioner.load_partition(0)
+        p1 = partitioner.load_partition(1)
+        all_cols = p0.column_names + p1.column_names
+        self.assertNotIn("drop_me", all_cols)
+
+    def test_partitioning_with_shared_columns(self) -> None:
+        """Check shared columns added to every partition."""
+        columns = ["f1", "f2", "shared"]
+        dataset = _create_dummy_dataset(columns)
+        partitioner = VerticalSizePartitioner(
+            partition_sizes=[1, 1], shared_columns=["shared"], shuffle=False
+        )
+        partitioner.dataset = dataset
+        p0 = partitioner.load_partition(0)
+        p1 = partitioner.load_partition(1)
+        self.assertIn("shared", p0.column_names)
+        self.assertIn("shared", p1.column_names)
+
+    def test_partitioning_with_active_party_add_to_last(self) -> None:
```

Comment:
this test and the one below needs to be revisited given the change in how `partition_sizes` expects `list[int]` (they now need to add up to the number of columns).

Reply:
I've added new tests now but also updated the size check which was incorrent.